doc/python/sphinx/api.rst

```diff
@@ -0,0 +1,123 @@
+API Reference
```

Comment:
I list the sections there because I saw the interfaces was separated into those sections in code. And my goal is wanting user to easily navigate through the documentation. The paradigm I am following is the documentation of [Flask](http://flask.pocoo.org/docs/1.0/api/). It uses sections to separate APIs into trunks.

How about following aggregation? Spliting `Functions` and `gRPC Enums` into:
* Create Client
* Create Client Credentials
* Create Server
* Create Server Credentials
* RPC Method Handlers
* Channel Ready Future
* Channel Connectivity
* gRPC Status Code

Reply:
> I list the sections there because I saw the interfaces was
> separated into those sections in code.

Yeah, but I'm saying _it's not a good thing that they're separated into those sections in code_. I feel that that section-division is _bad_, and it should not be used as an example.

One of the things that's done in the code that's good and perhaps worth repeating in the reference is that all the code elements are in [definitions-before-uses order](https://www.youtube.com/watch?v=aOEfIrC07XA&t=27m21s). Did you notice that? Do you have any particular feelings about it? Something that I definitely don't like about the current order is the way the reader can see a function that returns a `grpc.Future` and then ask "well what's a `grpc.Future`?" and not see the specification of `grpc.Future` until all the way at the bottom.

An important question we should ask is whether this document is a _tutorial_ or a _reference_. I'm pretty sure that in its current state it's a reference. If we wanted it to be a tutorial, we'd have to add a lot more prose and examples (and besides, we already have tutorials on [grpc.io](https://grpc.io)! Something important about _reference_ documentation is that it's not really meant to be read start to finish; users are expected that they'll just jump to whatever element it is about which they need to read. So the page could even be in alphabetical order.

Here's another thing to consider: the sections currently don't have any prose attached to them. They don't have any sentences describing their area of concern or any sort of larger ideas. That's a sign that maybe their existence isn't really justified.